native/cocos/2d/renderer/Batcher2d.cpp

```diff
@@ -94,9 +95,17 @@ void Batcher2d::syncRootNodesToNative(ccstd::vector<Node*>&& rootNodes) {
 }
 
 void Batcher2d::fillBuffersAndMergeBatches() {
+    size_t index = 0;
     for (auto* rootNode : _rootNodeArr) {
         walk(rootNode, 1);
         generateBatch(_currEntity, _currDrawInfo);
+
+        auto const scene = rootNode->getScene()->getRenderScene();
+        size_t const count = _batches.size();
+        for (size_t i = index; i < count; i++) {
+            scene->addBatch(_batches.at(i));
+        }
+        index = count;
```

Comment:
Why assign to count?

Reply:
Cause want to add batch segment generated by the current root to the renderscene of the current root